File: 2st/Yeonwoo/소인수분해.py
def solution(n):
    answer = []
    temp = []
    for i in range(1,n+1):
        if n%i==0:
            temp.append(n//i)
    temp.sort()
    for j in temp:
        cnt = 0
        for k in range(1,j+1):
            if j%k==0:
                cnt+=1
        if cnt==2:
            answer.append(j)
    return answer

File: 4st/Eunseong/소인수분해.java
class Solution {
    public int[] solution(int n) {
        int cnt = 0;
        boolean isSosu = true;
        int m = n;
        for (int i = 2; i <= m; i++) {
            isSosu = true;
            for (int j = 2; j < i; j++) {
                if (i % j == 0) {
                    isSosu = false;
                    break;
                }
            }
            if (isSosu) {
                if (n % i == 0) cnt++;
                while (m % i == 0) {
                    m /= i;
                }
            }
        }
        int[] answer = new int[cnt];
        cnt = 0;
        for (int i = 2; i <= n; i++) {
            isSosu = true;
            for (int j = 2; j < i; j++) {
                if (i % j == 0) {
                    isSosu = false;
                    break;
                }
            }
            if (isSosu) {
                if (n % i == 0) {
                    answer[cnt++] = i;
                    while (n % i == 0) {
                        n /= i;
                    }
                }
            }
        }
        return answer;
    }
}